flask/views.py

```diff
@@ -102,12 +102,16 @@ def view(*args, **kwargs):
         return view
 
 
+def get_methods(cls):
+    return getattr(cls, 'methods', []) or []
+
+
 class MethodViewType(type):
 
     def __new__(cls, name, bases, d):
         rv = type.__new__(cls, name, bases, d)
         if 'methods' not in d:
-            methods = set(rv.methods or [])
+            methods = set(m for b in bases for m in get_methods(b))
```

Comment:
That's is a good point, I'll add a doc fix for this in this pull request and ping you once this is done.